Lib/importlib/_bootstrap_external.py

```diff
@@ -496,16 +504,28 @@ def _compile_bytecode(data, name=None, bytecode_path=None, source_path=None):
         raise ImportError('Non-code object in {!r}'.format(bytecode_path),
                           name=name, path=bytecode_path)
 
-def _code_to_bytecode(code, mtime=0, source_size=0):
-    """Compile a code object into bytecode for writing out to a byte-compiled
-    file."""
+
+def _code_to_timestamp_pyc(code, mtime=0, source_size=0):
+    "Produce the data for a timestamp-based pyc."
     data = bytearray(MAGIC_NUMBER)
+    data.extend(_w_long(0))
```

Comment:
The reason the flags field comes first is that it controls the interpretation of rest of the header. So, it just made since to me to put the first fields you have to read first. Not the most compelling reason, but it's a very simple format after all!

I would prefer to solve the year-of-2038 problem by extending the whole header to 24 bytes to have 64-bit mtime and size fields. Letting the mtime "overflow" into the flags field by clever ordering fields doesn't seem worth it. The current code will probably even continue to work in 2038, since it explicitly only compares the lower 32-bits of the mtime. So, you'll only have problems if you have 68 year old source files!